app/services/pending_payments.rb

```diff
@@ -0,0 +1,15 @@
+# Returns the capturable payment object for an order with balance due
+
+class PendingPayments
+  def initialize(order)
+    @order = order
+  end
+
+  def payment_object
+    @order.payments.select{ |p| p if p.state == 'checkout' }.first
+  end
+
+  def can_be_captured?
+    payment_object && payment_object.actions.include?('capture')
+  end
+end
```

Comment:
IMO this is a model an not a service. It represents a concept of our domain and not a command to be performed. I'd move it to `model/`. Btw, really nice abstraction :ok_hand:

Reply:
You really think this should be a model and not a service? It's task is just "return the payable payment object for an order".

Reply:
To me, the key is that you named it after a noun and not a verb and you don't perform an action that changes the system's state.

Reply:
The purpose of this class is maybe not the modelling of pending payments. We could just call `order.pending_payments` for that. This service is working out if there is something to capture. There is not much logic to that though and I would probably get rid of the service all together and have it in the serialiser. It's shorter as well:

```ruby
  def capture_path
    pending_payment = object.pending_payments.first
    return '' unless object.payment_required? && pending_payment
    spree_routes_helper.fire_admin_order_payment_path(object, pending_payment.id, e: 'capture')
  end
```

Reply:
Done. That's much better.